pkg/scheduler/framework/plugins/volumebinding/volume_binding.go

```diff
@@ -299,7 +299,7 @@ func New(plArgs runtime.Object, fh framework.Handle) (framework.Plugin, error) {
 	if utilfeature.DefaultFeatureGate.Enabled(features.CSIStorageCapacity) {
 		capacityCheck = &scheduling.CapacityCheck{
 			CSIDriverInformer:          fh.SharedInformerFactory().Storage().V1().CSIDrivers(),
-			CSIStorageCapacityInformer: fh.SharedInformerFactory().Storage().V1alpha1().CSIStorageCapacities(),
+			CSIStorageCapacityInformer: fh.SharedInformerFactory().Storage().V1beta1().CSIStorageCapacities(),
```

Comment:
What happens if the scheduler is updated before the apiserver (or before 1 of the apiservers) and ends ups askig for beta which doesn't exist?  Is that a problem or will it recover?

Or do you need to leave this on v1alpha1 until next release, at which point you can switch?

Reply:
> What happens if the scheduler is updated before the apiserver (or before 1 of the apiservers) and ends ups askig for beta which doesn't exist? Is that a problem or will it recover?

This is indeed a bit problematic, depending on what is expected. kube-scheduler waits for all caches to sync (cmd/kube-scheduler/app/server.go), which fails for the beta API if not supported by the kube-apiserver. kube-scheduler then waits and does not schedule anything. It will recover once the apiserver also gets updated. Is that acceptable?

Just to be sure that this really recovers, I tried it:
- deploy cluster with v1alpha1 API
- update kube-scheduler with v1beta1 API
- create pod with unbound volume, late binding and storage capacity tracking -> stuck, as expected
- update apiserver with v1beta1 API
- kube-scheduler gets the storage capacity information, pod starts

> Or do you need to leave this on v1alpha1 until next release, at which point you can switch?

Using v1alpha1 wouldn't help: in a normal cluster, that API group won't be available in kube-apiserver by default, so updating kube-scheduler before kube-apiserver will still stop scheduling.

But is this really a problem? https://kubernetes.io/docs/tasks/administer-cluster/cluster-upgrade/#manual-deployments explicitly instructs to upgrade etcd and kube-apiserver first. If admins don't follow that advice, some breakage has to be expected and isn't too bad in this case (will recover, log output about watch explains what's failing).

Reply:
> Or do you need to leave this on v1alpha1 until next release, at which point you can switch?

My other motivation for using the v1beta1 API was that I intended to add MaximumVolumeSize there, and only there. But that actually then caused test failures during roundtripping and munnerz convinced me that the new field should be added also to v1alpha1 (https://kubernetes.slack.com/archives/C0EG7JC6T/p1614956222111400?thread_ts=1614955402.110000&cid=C0EG7JC6T).

Therefore v1alpha1 and v1beta1 are now functionally equivalent and we could use it in kube-scheduler without loss of functionality. In case of version skew against an older kube-apiserver it simply wouldn't get the MaximumVolumeSize, which is fine.

But my point about the alpha API group not being enabled still stands. Only clusters where the CSIStorageCapacity API was already enabled prior to 1.21 will be able to use kube-scheduler 1.21. Is that enough reason to change the PR or should I keep it as-is, i.e. use v1beta1 in kube-scheduler?

Reply:
Since apiserver needs to be upgraded first, I think it is fine to switch the scheduler to use v1beta1 (and the fact that v1alpha1 needs explicit opt-in)